src/main/java/amazin/service/AmazonService.java

```diff
@@ -0,0 +1,83 @@
+package amazin.service;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+@Service
+public class AmazonService {
+
+    private AmazonS3 s3client;
+
+    @Value("${amazonProperties.endpointUrl}")
+    private String endpointUrl;
+    @Value("${amazonProperties.bucketName}")
+    private String bucketName;
+    @Value("${amazonProperties.accessKey}")
+    private String accessKey;
+    @Value("${amazonProperties.secretKey}")
+    private String secretKey;
+
+    @PostConstruct
+    private void initializeAmazon() {
+        AWSCredentials credentials = new BasicAWSCredentials(this.accessKey, this.secretKey);
+        s3client = AmazonS3ClientBuilder
+                    .standard()
+                    .withRegion("us-east-1")
+                    .withCredentials(new AWSStaticCredentialsProvider(credentials)).build();
+    }
+
+    private String generateFileName(File file) {
+        return "pictures/" + new Date().getTime() + "-" + file.getName().replace(" ", "_");
+    }
+
+    private void uploadFileTos3bucket(String fileName, File file) {
+        PutObjectRequest req = new PutObjectRequest(bucketName, fileName, file)
+                .withCannedAcl(CannedAccessControlList.PublicRead);
+        s3client.putObject(req);
+    }
+
+    private File convertMultiPartToFile(MultipartFile file) throws IOException {
+        File convFile = new File(file.getOriginalFilename());
+        FileOutputStream fos = new FileOutputStream(convFile);
+        fos.write(file.getBytes());
+        fos.close();
+        return convFile;
+    }
+
+    public String uploadFile(MultipartFile multipartFile) {
+
+        String fileUrl = "";
+        try {
+            File file =  convertMultiPartToFile(multipartFile);
+            String fileName = generateFileName(file);
+            fileUrl = endpointUrl + "/" + bucketName + "/" + fileName;
+            uploadFileTos3bucket(fileName, file);
+            file.delete();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
might want to get this error back to the user if it fails instead of just printing the stack trace, e.g. image too large - do we have a limit on the image size right now?

Reply:
It's best to set a restriction on the image to prevent an attacker from uploading a file that would fill the HD and crash the server. For now, let's say each image should not exceed 2mb. Also we should only allow images to be uploaded therefore we we should blacklist all other file extensions except for JPEG and PNG